routes/actions.js

```diff
@@ -361,4 +366,19 @@ function returnToPath(req, res) {
     returnTo = '/';
   res.redirect(returnTo);
 }
-module.exports = router;
\ No newline at end of file
+
+// check for signupLanguage, ensure valid lang
+// set locale cookie if conditions met, else do nothing
+function setSignupLanguage(req, res) {
+    if (req.body.signupLanguage && languages.isValid(req.body.signupLanguage)) {
```

Comment:
Nit: `if` block should be indented. (I tend to use a beautifier to do this kind of clean-up automatically within the editor; I recommend http://atom.io/ . It also lets you use ESLint, which will flag violations of the style rules defined for this repository.)